accepted/future-releases/inline-classes/feature-specification.md

```diff
@@ -0,0 +1,969 @@
+restriction is that it is unsound to execute code in the body of `V` in
+the case where the values of the type variables do not satisfy their
+declared bounds, and those values will be obtained directly from the static
+type of the receiver in each member invocation on `V`.*
+
+When `s` is zero,
+<code>V&lt;T<sub>1</sub>, .. T<sub>s</sub>&gt;</code>
+simply stands for `V`, a non-generic inline type.
+When `s` is greater than zero, a raw occurrence `V` is treated like a raw
+type: Instantiation to bound is used to obtain the omitted type arguments.
+*Note that this may yield a super-bounded type, which is then a
+compile-time error.*
+
+We say that the static type of said variable, parameter, etc.
+_is the inline type_
+<code>V&lt;T<sub>1</sub>, .. T<sub>s</sub>&gt;</code>,
+and that its static type _is an inline type_.
+
+A compile-time error occurs if an inline type is used as a
+superinterface of a class or a mixin, or if an inline type is used to
+derive a mixin.
+
+*In other words, an inline type cannot occur as a superinterface in an
+`extends`, `with`, `implements`, or `on` clause of a class or mixin.
+On the other hand, it can occur in other ways, e.g., as a type
+argument of a superinterface of a class etc.*
+
+If `e` is an expression whose static type `V` is the inline type
+<code>Inline&lt;T<sub>1</sub>, .. T<sub>s</sub>&gt;</code>
+and `m` is the name of a member declared by `V`, then a member access
+like `e.m(args)` is treated as
+<code>invokeInlineMethod(Inline, &lt;T<sub>1</sub>, .. T<sub>s</sub>&gt;, e).m(args)</code>,
+and similarly for instance getters, setters, and operators.
+
+*In the body of an inline class declaration _DV_ with name `Inline`
+and type parameters
+<code>X<sub>1</sub>, .. X<sub>s</sub></code>, for an invocation like
+`m(args)`, if a declaration named `m` is found in the body of _DV_
+then that invocation is treated as
+<code>invokeInlineMethod(Inline, &lt;X<sub>1</sub>, .. X<sub>s</sub>&gt;, this).m(args)</code>.
+This is just the same treatment of `this` as in the body of a class.*
+
+*For example:*
+
+```dart
+extension E1 on int {
+  void foo() { print('E1.foo'); }
+}
+
+inline class V1 {
+  final int it;
+  V1(this.it);
+  void foo() { print('V1.foo'); }
+  void baz() { print('V1.baz'); }
+  void qux() { print('V1.qux'); }
+}
+
+void qux() { print('qux'); }
+
+inline class V2 {
+  final V1 it;
+  V2(this.it);
+  void foo() { print('V2.foo); }
+  void bar() {
+    foo(); // Prints 'V2.foo'.
+    it.foo(); // Prints 'V1.foo'.
+    it.baz(); // Prints 'V1.baz'.
+    1.foo(); // Prints 'E1.foo'.
+    1.baz(); // Compile-time error.
+    qux(); // Prints 'qux'.
+  }
+}
+```
+
+*That is, when the static type of an expression is an inline type `V`
+with representation type `R`, each method invocation on that
+expression will invoke an instance method declared by `V` or inherited
+from a superinterface (or it could be an extension method with on-type
+`V`).  Similarly for other member accesses.*
```

Comment:
Is the behavior of Object members called out anywhere?

Reply:
Yes.